src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/incentives/IncentivesResource.kt

```diff
@@ -240,4 +244,240 @@ class IncentivesResource(private val incentivesService: IncentivesService) {
+    requestBody = io.swagger.v3.oas.annotations.parameters.RequestBody(
+      content = [
+        Content(
+          mediaType = "application/json",
+          schema = Schema(implementation = CreateIncentiveRequest::class),
+        ),
+      ],
+    ),
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Updated Global Incentive level",
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Forbidden to access this endpoint when role NOMIS_INCENTIVES not present",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Global incentive level not found",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+    ],
+  )
+  fun updateGlobalIncentiveLevel(
+    @Schema(description = "Incentive reference code", example = "STD", required = true)
+    @PathVariable
+    code: String,
+    @RequestBody @Valid
+    updateIncentiveRequest: UpdateGlobalIncentiveRequest,
+  ): ReferenceCode =
+    incentivesService.updateGlobalIncentiveLevel(code, updateIncentiveRequest)
+
+  @PreAuthorize("hasRole('ROLE_NOMIS_INCENTIVES')")
+  @GetMapping("/incentives/reference-codes/{code}")
+  @ResponseStatus(HttpStatus.OK)
+  @Operation(
+    summary = "Gets the global incentive level by code",
+    description = "Gets a global incentive level by provided code and domain of IEP_LEVEL",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "return the Global Incentive level",
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Global Incentive Level does not exist",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+    ],
+  )
+  fun getGlobalIncentiveLevel(
+    @Schema(description = "Incentive reference code", example = "STD", required = true)
+    @PathVariable
+    code: String,
+  ): IEPLevel =
```

Comment:
How come create and update return a different type?

Reply:
The create and update return ReferenceCode.  I'll change the getter to return ReferenceCode (they are essentially the same thing)